apollo-router-core/src/query_planner/mod.rs

```diff
@@ -395,48 +356,108 @@ impl FetchNode {
+        let fetcher = service_registry.get(service_name).unwrap();
+
+        let (res, _tail) = fetcher
+            .stream(
+                Request::builder()
+                    .query(operation)
+                    .variables(Some(Arc::new(variables)))
+                    .build(),
+            )
+            .await
+            .into_future()
+            .instrument(query_span)
+            .await;
+
+        let subgraph_response = match res {
+            Some(response) if !response.is_primary() => {
+                return Err(FetchError::SubrequestUnexpectedPatchResponse {
+                    service: service_name.to_owned(),
+                });
+            }
+            Some(subgraph_response) => subgraph_response,
+            None => {
+                return Err(FetchError::SubrequestNoResponse {
                     service: service_name.to_string(),
-                }),
+                })
             }
+        };
+
+        self.response_at_path(current_dir, paths, subgraph_response)
+    }
+
+    fn response_at_path<'a>(
+        &'a self,
+        current_dir: &'a Path,
+        paths: Option<Vec<Path>>,
+        subgraph_response: Response,
+    ) -> Result<Value, FetchError> {
+        let Response { data, .. } = subgraph_response;
+
+        if self.requires.is_some() {
+            // we have to nest conditions and do early returns here
+            // because we need to take ownership of the inner value
+            if let Value::Object(mut map) = data {
+                if let Some(entities) = map.remove("_entities") {
+                    tracing::trace!(
+                        "Received entities: {}",
+                        serde_json::to_string(&entities).unwrap(),
+                    );
+
+                    if let Value::Array(array) = entities {
+                        let span = tracing::trace_span!("response_insert");
+                        let _guard = span.enter();
+
+                        let mut value = Value::default();
+
+                        let paths = paths.unwrap();
+                        for (entity, path) in array.into_iter().zip(paths.into_iter()) {
+                            let v = Value::from_path(&path, entity);
+                            value.deep_merge(v);
+                        }
+                        return Ok(value);
+                    } else {
+                        return Err(FetchError::ExecutionInvalidContent {
+                            reason: "Received invalid type for key `_entities`!".to_string(),
+                        });
+                    }
+                }
+            }
+
+            Err(FetchError::ExecutionInvalidContent {
+                reason: "Missing key `_entities`!".to_string(),
+            })
+        } else {
+            let span = tracing::trace_span!("response_insert");
+            let _guard = span.enter();
+
+            Ok(Value::from_path(current_dir, data))
```

Comment:
:screwdriver:  As per our ADR on telemetry, tracing's spans should be place on the source function and not the caller

Reply:
right, I was following the previous code on that, it's a good opportunity to change it

Reply:
done in 45c4df2